lib/c-api/src/wasm_c_api/externals/mod.rs

```diff
@@ -31,67 +165,41 @@ pub unsafe extern "C" fn wasm_extern_copy(r#extern: &wasm_extern_t) -> Box<wasm_
 pub unsafe extern "C" fn wasm_extern_delete(_extern: Option<Box<wasm_extern_t>>) {}
 
 #[no_mangle]
-pub unsafe extern "C" fn wasm_func_as_extern(
-    func: Option<&wasm_func_t>,
-) -> Option<Box<wasm_extern_t>> {
-    let func = func?;
-
-    Some(Box::new(wasm_extern_t {
-        instance: func.instance.clone(),
-        inner: Extern::Function(func.inner.clone()),
-    }))
+pub unsafe extern "C" fn wasm_func_as_extern(func: Option<&wasm_func_t>) -> Option<&wasm_extern_t> {
+    std::mem::transmute::<Option<&wasm_func_t>, Option<&wasm_extern_t>>(func)
 }
```

Comment:
I woud prefer to get `pub extern "C" fn…` here, and an `unsafe { … }` block inside the funciton, with a comment that explains why it's safe to transmute in this case :-).

We may want to add a `debug_assert!` to ensure the size/layout is correct between the both?